libs/libc/stdio/lib_freopen.c

```diff
@@ -100,9 +102,19 @@ FAR FILE *freopen(FAR const char *path, FAR const char *mode,
           return NULL;
         }
 
-      /* Flush the stream and duplicate the new fd to it */
+      /* Get the new file position */
+
+      f_pos = lseek(fd, 0, SEEK_CUR);
+
+      /* Flush the stream and set the file position. The later is needed to
+       * ensure that the read buffers are also flushed.
+       */
 
       fflush(stream);
+      fseek(stream, f_pos, SEEK_SET);
```

Comment:
should we add a new function to simply discard the buffer?

Reply:
I will change it to directly call `lib_rdflush()`.